pkgs/native_toolchain_c/lib/src/tool/tool_resolver.dart

```diff
@@ -208,11 +208,12 @@ class InstallLocationResolver implements ToolResolver {
   Future<List<Uri>> tryResolvePath(String path) async {
     if (path.startsWith(home)) {
```

Comment:
If it starts with `$HOME`, and `homeDir_` is null, shouldn't we simply `return [];` early?

Reply:
I was wondering why this could ever be null anyway, it should not be.

https://github.com/dart-lang/native/pull/2112

Reply:
Do you still want to null check this? or should we close it? Maybe an env issue could still cause null here.

Reply:
Yeah it wouldn't hurt to check and return `[]` if there is no home. Then at least the code doesn't crash if the home path environment variable isn't set. 👍

Reply:
Cool, should be all set now.

Reply:
Thanks for your contributions @MichealReed ! 🙏